snoop/data/digests.py

```diff
@@ -117,17 +118,24 @@ def gather(blob, **depends_on):
                 rv['ocrimage'] = True
         rv['ocrtext'] = ocr_results
 
-    # combine texts and detect language
-    if settings.DETECT_LANGUAGE:
+    #TODO: Add entity Extraction setting to snoop settings
+    if settings.EXTRACT_ENTITIES:
+        text = rv.get('text', '')
+        ocrtexts = [v for v in rv.get('ocrtext', {}).values() if v]
+        if text or ocrtexts:
+            alltext = text + "\n" + "\n".join(ocrtexts)
```

Comment:
Don't concatenate these texts.

For language detection concatenate first 3K of each one like below, and for NER do it separately per text, as explained below.